src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -396,4 +411,62 @@ public PIDController getYController() {
     public PIDController getRotationController() {
         return rotationController;
     }
+
+    // Adds the current pos to the latency list
+    public void updateLatencyList() {
+        // add timestamp pair
+        double currentTime = Timer.getFPGATimestamp();
+        pastPositions.add(new Pair<>(currentTime, getPose()));
+
+        // remove first until maxSecondsStored is met
+        while (pastPositions.size() > 0 && currentTime - pastPositions.get(0).getFirst() > maxSecondsStored) {
+            pastPositions.remove(0);
+        }
+    }
+
+    // util for latency calculation
+    // taken from https://www.geeksforgeeks.org/java-equivalent-of-cpp-lower_bound-method/ (slightly modified)
```

Comment:
delete this line. its nice to add source links but this should be in documentation for vision, not in the actual code.